### PR TITLE
DDSQL Reorg

### PR DESCRIPTION
**Pull request by estherk15**
### What does this PR do? What is the motivation?

- Merge Aggregations and Scalar Functions pages
- Move all reference docs out one level in navigation and remove Reference landing page
- Move images from `dashboards` to `ddsql_editor` folder
- Update the nav menu
- Update reference links
- Note: left the folder structure alone because of [CODEOWNERS](https://github.com/DataDog/documentation/blob/836dd9b4cbe2b68efbe27cbaa0ca0917c0f89546/.github/CODEOWNERS#L155) file, all reference files are listed
- PM Requested [DOCS-8628](https://datadoghq.atlassian.net/browse/DOCS-8628)

### Merge instructions

- [x] Please merge after reviewing

### Additional notes

[DOCS-8628]: https://datadoghq.atlassian.net/browse/DOCS-8628?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/ddsql_editor/_index.md [hunk 1]
```diff
@@ -39,29 +40,37 @@ WHERE env = 'staging' -- env is a tag, not a column
...
 
 View and filter the list of tables and fields in the schema side panel:
 
-{{< img src="dashboards/ddsql/schema-explorer.png" alt="A list of available tables" style="width:90%;" >}}
+{{< img src="ddsql_editor/schema-explorer.png" alt="A list of available tables" style="width:90%;" >}}
 
 Click a table name to view its columns and relationships:
 
-{{< img src="dashboards/ddsql/table-details.png" alt="The details of a table, including its columns and relationships" style="width:60%;" >}}
+{{< img src="ddsql/table-details.png" alt="The details of a table, including its columns and relationships" style="width:60%;" >}}
```
Comment (alai97) [inline content/en/ddsql_editor/_index.md:59]:
```suggestion
{{< img src="ddsql_editor/table-details.png" alt="The details of a table, including its columns and relationships" style="width:60%;" >}}
```

File: content/en/ddsql_editor/_index.md [hunk 2]
```diff
@@ -39,29 +40,37 @@ WHERE env = 'staging' -- env is a tag, not a column
...
 
 Save useful queries, or export the data as CSV.
 
-{{< img src="dashboards/ddsql/save-or-export-result.png" alt="A query result with the save and export actions shown" style="width:90%;" >}}
+{{< img src="ddsql_editor/save-or-export-result.png" alt="A query result with the save and export actions shown" style="width:90%;" >}}
 
 Browse and re-run saved queries in the side panel.
 
-{{< img src="dashboards/ddsql/saved-queries-panel.png" alt="A list of saved queries" style="width:60%;" >}}
+{{< img src="ddsql/saved-queries-panel.png" alt="A list of saved queries" style="width:60%;" >}}
```
Comment (alai97) [inline content/en/ddsql_editor/_index.md:69]:
```suggestion
{{< img src="ddsql_editor/saved-queries-panel.png" alt="A list of saved queries" style="width:60%;" >}}
```

**Review by alai97 — COMMENTED** [review]
Looks good, two image folder renaming nits!

File: content/en/ddsql_editor/_index.md (hunk 1 above)
Comment (estherk15) [reply content/en/ddsql_editor/_index.md:59]: Good catch!

**Review by alai97 — APPROVED** [review]

**Pull request merged**